### ISSUE TITLE
Predicting from polynomial fit to frequency axis

### ISSUE THREAD

**Issue by landmanbester** [ska-sa/montblanc#171]
Currently montblanc uses a spectral index model for the frequency axis

I(v) = I(v0)(v/v0)^alpha

In some cases it is preferable to fit a more flexible model. In particular it is not possible to fit in log space when there are negative pixels in the model. In such cases we can still fit a polynomial

I(v) = I_0 + I_1(v/v0) +   I_2(v/v0)**2 + ....    

to the frequency axis. Alternatively we might want to fit a polynomial in log space

log(I(v)) = I_0 + alpha log(v/v0) +   beta log(v/v0)**2 + ....    

Can we have a feature which allows montblanc to predict from these models? This is especially relevant for incorporating montblanc into DDFacet.

**Comment by landmanbester** [comment]
This is also relevant to https://github.com/ska-sa/montblanc/issues/169

**Comment by landmanbester** [comment]
Just documenting discussion with @sjperkins. We usually want to fit the polynomial in log space for Stokes I and just an ordinary polynomial for Q, U and V. Thus sensible default behaviour would be to use a log-polynomial for I and ordinary polynomial for the rest. There are however cases where we want to allow for negative flux (as is the case when predicting from a DDFacet sky model for example) so we should be able to toggle log-polynomial on or off depending on the use case.

**Comment by sjperkins** [comment]
The `alpha` input with shape schema `['nsrc', 'ntime']` would then be renamed to `spectral_components` and it's shape changed something like  `['nsrc','ntime','nspeccomps','nstokes']`.

@o-smirnov Is the variation by time necessary -- The original idea was to represent scintillation in the `stokes` parameters, but I've never been sure if it's necessary in the spectral index expression.

**Comment by rdeane** [comment]
The spectral index (and stokes parameters) can change fairly rapidly in the
EHT case (timescale of minutes to hours). This is an intrinsic source
variation which we would like to model in the future, so if this can be
incorporated now with not much extra effort, I would be greatly
appreciative.

cheers
Roger

**Comment by bennahugo** [comment]
I second @landmanbester's request to do polynomial fitting for I. It is necessary if you're working off pixelated models that could contain negatives - if a source falls between pixels it is necessary to to describe it as some sort of sinc function in pixel space.

I also agree that the log expansion needs more terms - some sources are peaked within the observed spectrum and have strong curavature (1934-638 is one such case for instance).

Moreover it is **not** just one polynomial to describe Q and U behaviour in frequency. There are two taylor expansions needed to describe how the polarization angle varies with frequency and another how fractional polarization varies with frequency. In the a first order these are the polarization angle and polarization fraction at the reference frequency. See also documentation of SetJy here: https://casa.nrao.edu/docs/taskref/setjy-task.html

**Comment by sjperkins** [comment]
@landmanbester @bennahugo Am I correct in understanding that these expressions are applied to the stokes parameters prior to the construction of the brightness matrix?

**Comment by bennahugo** [comment]
Yes they make I, Q and U frequency dependent. So the brightness matrix becomes frequency dependent.

**Comment by sjperkins** [comment]
I'll work this into the dask version

**Comment by sjperkins** [comment]
I'll work this into the dask version
